File: src/main/java/pl/yalgrin/playnite/simplesync/repository/FeatureRepository.java
package pl.yalgrin.playnite.simplesync.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.stereotype.Repository;
import pl.yalgrin.playnite.simplesync.domain.Feature;
import reactor.core.publisher.Flux;

@Repository
public interface FeatureRepository extends ObjectRepository<Feature> {
    @Query("select c.id from playnite_feature c order by c.id asc")
    Flux<Long> findAllIds();
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/repository/FeatureRepository.kt
package pl.yalgrin.playnite.simplesync.library.repository

import org.springframework.data.r2dbc.repository.Query
import org.springframework.stereotype.Repository
import pl.yalgrin.playnite.simplesync.library.domain.Feature
import reactor.core.publisher.Flux

@Repository
interface FeatureRepository : ObjectRepository<Feature> {
    @Query("select c.id from playnite_feature c order by c.id")
    override fun findAllIds(): Flux<Long>
}
